backend/routers/social_media_marketing_router.py

```
@@ -828,57 +828,104 @@ async def get_swamiji_avatar_config(request: Request, admin_user: dict = Depends
+        logger.error(f"Error generating single avatar preview: {e}", exc_info=True)
         raise HTTPException(
-            status_code=500,
-            detail=f"Preview generation failed: {str(e)}"
-        ) from e
+            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+            detail=str(e)
+        )
+
+@social_marketing_router.post("/generate-all-avatar-previews", response_model=StandardResponse, tags=["Social Media Marketing", "Admin"])
+async def generate_all_avatar_previews(
+    request: GenerateAllAvatarPreviewsRequest,
```

Ruff (0.12.2)

Undefined name `GenerateAllAvatarPreviewsRequest`

(F821)